packages/webui/src/components/Button.tsx

```diff
@@ -23,15 +25,45 @@ function Button(
     <button
       ref={finalRef}
       className={tw(
-        'p-2 text-secondary bg-primary border border-solid border-primary rounded-md shadow-sm hover:bg-gray-50',
-        border === 'ghost' && 'bg-transparent border-transparent shadow-none',
-        size === 'small' && 'text-sm h-7 p-1',
-        size === 'large' && 'h-10',
+        // layout
+        'inline-flex items-center gap-x-1.5 p-2',
+        !!children && 'px-3',
+
+        // common styles
+        'ring-1 ring-inset rounded-md font-bold shadow-sm uppercase',
+        border === 'none' && 'ring-0 shadow-none',
+
+        // base colors
+        'text-manatee-900 ring-manatee-600 bg-manatee-100',
+        border === 'none' && 'bg-transparent',
+
+        // disabled
+        'disabled:text-manatee-400 disabled:ring-manatee-300 disabled:bg-manatee-100',
+        border === 'none' && 'bg-transparent',
+
+        // focus
+        'focus:text-apple-900 focus:ring-apple-600 focus:bg-apple-200',
+
+        // hover
+        'hover:text-apple-900 hover:ring-[#2D7124] hover:bg-apple-200',
+
+        // pressed
+        'active:text-apple-950 active:ring-apple-500 active:bg-apple-500',
+
+        // selected
+        selected && 'text-apple-950 ring-apple-400 bg-apple-400',
+
+        // small
+        size === 'small' && 'p-1.5 text-xs',
+        size === 'small' && !!children && 'px-2',
+
+        // externals
```

Comment:
love the comments here to help make it clear what styles are part of what variant!